sched/scheduler/stateful_scheduler.go

```diff
@@ -92,11 +105,11 @@ func NewStatefulScheduler(
 
 	// TODO: we need to allow the scheduler to accept new jobs
 	// while recovering old ones.
-	if recoverJobsOnStartup {
+	if config.RecoverJobsOnStartup {
 		sched.startUp()
 	}
 
-	if !debugMode {
+	if !config.DebugMode {
```

Comment:
exposing a debug mode flag in the config for automated (non-user interactive) testing seems a bit off to me....  if we need automated testing to have control over the progress of the scheduler loop couldn't we have injected instrumented workers where the test control's when a worker 'finishes'.  That way we don't have to put the sched.loop() in an 'if !debug mode {}' clause (change the code behavior for debug vs non-debug).... if we plan to expose the debug feature to the user then step() needs to be Step(), Step() should be in the API and the ScheduleJob() request would have the debug flag passed in on the request.